node/status_node.go

```diff
@@ -186,10 +195,13 @@ func (n *StatusNode) startPeerPool() error {
 		peers.NewCache(n.db),
 		options,
 	)
-	if err := n.register.Start(n.gethNode.Server()); err != nil {
+	if err := n.discovery.Start(); err != nil {
+		return err
+	}
+	if err := n.register.Start(n.discovery); err != nil {
```

Comment:
Should it stop discovery here as well? Or this error is supposed to shut down node anyway?

Reply:
we don't stop other things if error happened in the middle, so i assume that stopping a node on error is the only thing we will do anyway